src/app/UploadPilets.tsx

```diff
@@ -29,21 +56,59 @@ export const UploadPilets: FC<UploadPiletsProps> = () => {
           authorization: `Basic ac6c202085f07099da1729a20e5750e651ef093ef4a5856c70997a6cc71dcab2`,
         },
       })
-        .then(res => res.json())
-        .then(res => res.success && injectPiletsFromUrl(`${url}?id=${res.name}`));
+        .then(res => {
+          if (res.status === 400) {
+            setUploadData({
+              ...uploadData,
+              progress: 0,
+            });
+            throw res;
+          }
+          return res.json();
+        })
+        .then(res => {
+          res.success && injectPiletsFromUrl(`${url}?id=${res.name}`);
+          setUploadData({
+            ...uploadData,
+            progress: 0,
+            piletName: res.name,
+          });
+        })
+        .catch(err => {
+          err.text().then(errorMassage => {
```

Comment:
This is wrong - since you catch *after* the previous `then` you don't know what you caught.

I recommend using the failure branch of `then`instead.

```
x.then(success, failure)
```

instead of

```
x.then(success).catch(failure)
```